### ISSUE TITLE
📎 Parse `v-for` directives more precisely.

### ISSUE THREAD

**Issue by dyc3** [biomejs/biome#8062]
### Description

Vue's [`v-for`](https://vuejs.org/guide/essentials/list) directive has specific syntax that it uses in it's initializer.

```vue
<li v-for="item in items">
  {{ item.message }}
</li>
```

The HTML parser currently treats what's in the quotes as a generic string. However, some lint rules (like [`valid-v-for`](https://eslint.vuejs.org/rules/valid-v-for.html)) require inspecting the contents of this string for better analysis.

**Comment by Hebilicious** [comment]
There's also a small bug where the v-for of syntax errors, here's a [playground link](https://biomejs.dev/playground/?code=PABkAGkAdgAgAHYALQBmAG8AcgA9ACIAaQB0AGUAbQAgAGkAbgAgAGkAdABlAG0AcwAiAD4APAAvAGQAaQB2AD4ACgA8AGQAaQB2ACAAdgAtAGYAbwByAD0AIgBpAHQAZQBtACAAbwBmACAAaQB0AGUAbQBzACIAPgA8AC8AZABpAHYAPgA%3D&language=vue).

You can find the reference for vue for of here https://vuejs.org/guide/essentials/list.html#v-for (it's exactly the same as for in)
Both these should work : 

```vue
<div v-for="item in items"></div>
<div v-for="item of items"></div>
```
I can open a dedicated issue if needed.

**Comment by ematipico** [comment]
> There's also a small bug where the v-for of syntax errors, here's a [playground link](https://biomejs.dev/playground/?code=PABkAGkAdgAgAHYALQBmAG8AcgA9ACIAaQB0AGUAbQAgAGkAbgAgAGkAdABlAG0AcwAiAD4APAAvAGQAaQB2AD4ACgA8AGQAaQB2ACAAdgAtAGYAbwByAD0AIgBpAHQAZQBtACAAbwBmACAAaQB0AGUAbQBzACIAPgA8AC8AZABpAHYAPgA%3D&language=vue).
> 
> You can find the reference for vue for of here [vuejs.org/guide/essentials/list.html#v-for](https://vuejs.org/guide/essentials/list.html#v-for) (it's exactly the same as for in) Both these should work :
> 
> <div v-for="item in items"></div>
> <div v-for="item of items"></div>
> I can open a dedicated issue if needed.

I checked the page you shared, and I **can't find any examples** with `item of items`

**Comment by wvffle** [comment]
It's att the very bottom of the linked section:

<img width="1092" height="234" alt="Image" src="https://github.com/user-attachments/assets/190070cb-6f58-43ab-a9da-0c47d564892a" />
